wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/SyntaxRules.kt

```diff
@@ -22,6 +22,7 @@ import com.squareup.wire.schema.ProtoFile.Syntax.PROTO_3
 /** A set of rules which defines schema requirements for a specific [Syntax]. */
 interface SyntaxRules {
   fun enumRequiresZeroValueAtFirstPosition(): Boolean
+  fun extensionIsPermittedOnlyForCustomOptions(): Boolean
```

Comment:
How about turning it upside down and name it `allowsMessageExtensions()`? Plus, we're testing elsewhere, not here, if the type is an option so I'd rather take it out of the method name. I also wanna make it easy to read and understand.